Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -695,6 +696,42 @@ open class CarPlayNavigationViewController: UIViewController, BuildingHighlighti
         navigationMapView?.showWaypoints(on: progress.route, legIndex: legIndex)
     }
     
+    func updateLegacyIcons() {
+        spriteRepository.legacyCache.clearMemory()
+        updateManeuvers(navigationService.routeProgress)
+    }
+    
+    func updateInstruction(_ routeProgress: RouteProgress) {
+        guard spriteRepository.getSpriteImage() == nil else {
+            updateManeuvers(routeProgress)
+            return
+        }
+        
+        spriteRepository.updateSprite(styleURI: spriteRepository.styleURI) { [weak self] in
+             guard let self = self else { return }
+             self.updateManeuvers(self.navigationService.routeProgress)
+         }
+    }
+    
+    func carPlayManeuverLabelAttributedText(bounds: @escaping () -> (CGRect),
```

Comment:
Why did you decide to introduce this method This is essentially a copy of `carPlayManeuverLabelAttributedText(bounds:shieldHeight:window:instructionLabelType:)` (besides setting `instructionDelegate`).

Reply:
I do not recall why `instructionLabelType` was added in original method, but it was likely for a reason. With new method type information is removed, because all objects will be of type `InstructionLabel`, instead of either `PrimaryLabel` or `SecondaryLabel`.

Reply:
The method is introduced to be included in the `CarPlayNavigationViewController`, because it could use the `SpriteRepository` instance that cached large Sprite sheet ready to use. Otherwise when we use the original one we need to create a `SpriteRepository` and start downloading process again.

The `instructionLabelType` hasn't been used to generate the `attributedText` for the maneuver. We do have different style appearance for `PrimaryLabel` and `SecondaryLabel` in mobile top controller during active navigation, but they only have the font size difference. And we didn't apply them to the CarPlay before.

Reply:
Not sure I understand, this method doesn't use `SpriteRepository`. Reason I'm hesitant about including new method is that we're essentially leaving previous method unused (it's also public).

Reply:
This method doesn’t use `SpriteRepository`, because the PR add the `instructionLabel.instructionDelegate` to the `CarPlayNavigationViewController`, it would allow the `CarPlayNavigationViewController` to use its `SpriteRepository` to generate the `attributedText` for the `instructionLabel`.

But I removed this method, and used the `VisualInstruction.carPlayManeuverLabelAttributedText(bounds:shieldHeight:window:instructionLabelType:)`, by adding the `delegate` parameter to pass the `CarPlayNavigationViewController` to it.